packages/ui/primitives/document-flow/add-fields.tsx

```diff
@@ -566,7 +566,13 @@ export const AddFieldsFormPartial = ({
           onGoBackClick={() => {
             previousStep();
             remove();
+            documentFlow.onBackStep?.();
           }}
+          goBackLabel={
+            currentStep === 1 && typeof documentFlow.onBackStep === 'function'
+              ? 'Remove'
+              : undefined
+          }
```

Comment:
This wouldn't work since it would also affect the main webapp where the document wouldn't be removed 🤔

Reply:
I did a global lookup of where all `AddFieldsFormPartial` is used and it looks like it is used only in singleplayermode and `edit-document.tsx` which is a part of main webapp dashboard. The property `documentFlow.fields` in `edit-document.tsx` doesn't have the `onBackStep` property implemented, meaning `typeof documentFlow.onBackStep` would be falsy, thereby the default fallback string would be rendered (Please correct me If I'm wrong here). Nonetheless, I've made the changes by adding an additional optional prop to the `AddFieldsFormPartial` component to keep it future-proof. Kindly check.